no-useless-escape false positives on JSX strings

This is a JSX issue, but since it's in `eslint/no-useless-escape` I thought it would be appropriate to file here.

**What version of ESLint are you using?**
2.8.0

**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESLint

**Please show your full configuration:**
It's super long but these should be enough,
```js
module.exports = {

  // base off Airbnb Javascript Style Guide
  // https://github.com/airbnb/javascript
  'extends': [
    'eslint-config-airbnb',
  ],

  'parser': 'babel-eslint',
}
```

**What did you do? Please include the actual source code causing the issue.**
```jsx
<input
  autoComplete="cc-cvc cc-csc"
  maxLength="4"
  pattern="[\d]{3,4}"
  placeholder="CVC"
/>
```

**What did you expect to happen?**

The pattern `"[\d]{3,4}"` would not be flagged as the backslash is not an escape character in this context. (It's a literal backslash, like in HTML.)

**What actually happened? Please include the actual, raw output from ESLint.**

```
  135:21  error  Unnecessary escape character: \d  no-useless-escape
```

I saw the spec: https://facebook.github.io/jsx/
JSXText seems to not have any escape sequence.

And I confirmed this with espree, too.
I will fix this.

```
C:\Users\t-nagashima.AD\Documents\GitHub\eslint [master]> cat test.js
<foo attr="\d"/>;
C:\Users\t-nagashima.AD\Documents\GitHub\eslint [master]> eslint -v
v2.8.0
C:\Users\t-nagashima.AD\Documents\GitHub\eslint [master]> eslint test.js --rule no-useless-escape:2 --parser-options "ecmaFeatures:{jsx:true}" --no-ignore --no-eslintrc

C:\Users\t-nagashima.AD\Documents\GitHub\eslint\test.js
  2:12  error  Unnecessary escape character: \d  no-useless-escape

✖ 1 problem (1 error, 0 warnings)

--- a/lib/rules/no-useless-escape.js
+++ b/lib/rules/no-useless-escape.js
@@ -96,6 +96,12 @@ module.exports = function(context) {
         var pattern = /\\[^\d]/g;
 
         if (typeof node.value === "string") {
+
+            // JSXAttribute doesn't have any escape sequence: https://facebook.github.io/jsx/
+            if (node.parent.type === "JSXAttribute") {
+                return;
+            }
+
             nodeEscapes = VALID_STRING_ESCAPES;
         } else if (node.regex) {
             nodeEscapes = VALID_REGEX_ESCAPES;

--- a/tests/lib/rules/no-useless-escape.js
+++ b/tests/lib/rules/no-useless-escape.js
@@ -46,7 +46,9 @@ ruleTester.run("no-useless-escape", rule, {
         "var foo = '\\v';",
         "var foo = '\\f';",
         "var foo = '\\\n';",
-        "var foo = '\\\r\n';"
+        "var foo = '\\\r\n';",
+        {code: "<foo attr=\"\\d\"/>", parserOptions: {ecmaFeatures: {jsx: true}}},
+        {code: "<foo attr='\\d'></foo>", parserOptions: {ecmaFeatures: {jsx: true}}}
     ],
 
     invalid: [
@@ -69,7 +71,11 @@ ruleTester.run("no-useless-escape", rule, {
                 {line: 1, column: 14, message: "Unnecessary escape character: \\a", type: "Literal"},
                 {line: 1, column: 16, message: "Unnecessary escape character: \\@", type: "Literal"}
             ]
+        },
+        {
+            code: "<foo attr={\"\\d\"}/>",
+            parserOptions: {ecmaFeatures: {jsx: true}},
+            errors: [{ line: 1, column: 13, message: "Unnecessary escape character: \\d", type: "Literal"}]
         }
-
     ]
 });

